src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,47 @@
+import Popup from "./Popup.js";
+
+class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  open(cardData) {
+    super.open();
+    this._cardData = cardData;
+  }
+
+  close() {
+    this._popupForm.reset();
+    super.close();
```

Comment:
Please, delete method `close` from `PopupWithConfirmation` because you don't need it (you don't have the inputs)

Reply:
completed.